src/components/charts/SwipeChart.tsx

```diff
@@ -196,59 +195,54 @@ const SwipeChartComponent = (params: Props) => {
           sMinMaxOpacity.value = withDelay(ANIMATION_DURATION.maxMinFadeInDelay, withTiming(1, { duration: ANIMATION_DURATION.maxMinFadeIn }))
         }
 
-        try {
-          if (cachedChartData != null) {
-            // The chart price line animation is slow when transitioning directly
-            // between datasets.
-            // Add a delay so the component can get re-mounted with fresh data
-            // instead.
-            setTimeout(() => {
-              setChartData(cachedChartData)
-              setIsLoading(false)
-              delayShowMinMaxLabels()
-            }, 10)
-          } else {
-            const unixNow = Math.trunc(new Date().getTime() / 1000)
-            const fromParam = unixNow - queryFromTimeOffset
-            const fetchPath = sprintf(MARKET_CHART_ENDPOINT_4S, assetId, fiatCurrencyCode, fromParam, unixNow)
-            // Start with the free base URL
-            let fetchUrl = `${COINGECKO_URL}${fetchPath}`
-            do {
-              // Construct the dataset query
-              const response = await fetch(fetchUrl)
-              const result = await response.json()
-              const marketChartRange = asCoinGeckoMarketApi(result)
-              if ('status' in marketChartRange) {
-                if (marketChartRange.status.error_code === 429) {
-                  // Rate limit error, use our API key as a fallback
-                  if (!fetchUrl.includes('x_cg_pro_api_key')) {
-                    fetchUrl = `${COINGECKO_URL_PRO}${fetchPath}&x_cg_pro_api_key=${ENV.COINGECKO_API_KEY}`
-                    continue
-                  }
-                } else {
-                  throw new Error(JSON.stringify(marketChartRange))
+        if (cachedChartData != null) {
+          // The chart price line animation is slow when transitioning directly
+          // between datasets.
+          // Add a delay so the component can get re-mounted with fresh data
+          // instead.
+          setTimeout(() => {
+            setChartData(cachedChartData)
+            setIsLoading(false)
+            delayShowMinMaxLabels()
+          }, 10)
+        } else {
+          const unixNow = Math.trunc(new Date().getTime() / 1000)
+          const fromParam = unixNow - queryFromTimeOffset
+          const fetchPath = sprintf(MARKET_CHART_ENDPOINT_4S, assetId, fiatCurrencyCode, fromParam, unixNow)
+          // Start with the free base URL
+          let fetchUrl = `${COINGECKO_URL}${fetchPath}`
+          do {
+            // Construct the dataset query
+            const response = await fetch(fetchUrl)
+            const result = await response.json()
+            const marketChartRange = asCoinGeckoMarketApi(result)
+            if ('status' in marketChartRange) {
+              if (marketChartRange.status.error_code === 429) {
+                // Rate limit error, use our API key as a fallback
+                if (!fetchUrl.includes('x_cg_pro_api_key')) {
```

Comment:
Can you remove the dummy coingecko key from the config cleaner and add a check for it's existence here? Passing a dummy value to the api will produce a [10002 error](https://docs.coingecko.com/reference/common-errors-rate-limit) which is unhandled here and produces an error (an uglier error message with this commit's change). Instead, if the key doesn't exist we should just try again without an api key again.